x-pack/platform/plugins/shared/fleet/server/routes/agent/migrate_handlers.ts

```diff
@@ -12,14 +12,26 @@ import type {
   MigrateSingleAgentRequestSchema,
   BulkMigrateAgentsRequestSchema,
 } from '../../types';
+import { LICENSE_FOR_AGENT_MIGRATION } from '../../../common/constants';
 import * as AgentService from '../../services/agents';
+import { licenseService } from '../../services';
 
 export const migrateSingleAgentHandler: FleetRequestHandler<
   TypeOf<typeof MigrateSingleAgentRequestSchema.params>,
   undefined,
   TypeOf<typeof MigrateSingleAgentRequestSchema.body>
 > = async (context, request, response) => {
   const [coreContext] = await Promise.all([context.core, context.fleet]);
+
+  // Check the user has the correct license
+  if (!licenseService.hasAtLeast(LICENSE_FOR_AGENT_MIGRATION)) {
```

Comment:
What do you think of putting that check in the application service?  `kibana/x-pack/platform/plugins/shared/fleet/server/services/agents/migrate.ts` this way if we expose this to other plugins one day, we also have that license gate ?